src/Connection/Replication/SentinelReplication.php

```diff
@@ -538,7 +539,15 @@ protected function assertConnectionRole(NodeConnectionInterface $connection, $ro
     {
         $role = strtolower($role);
         $actualRole = $connection->executeCommand(RawCommand::create('ROLE'));
-
+        if (is_object($actualRole) && $actualRole instanceof Error) {
+            $info = $connection->executeCommand(RawCommand::create('INFO'));
+            preg_match('#role:([a-zA-Z]+)#m', $info, $matches);
```

Comment:
Hmm why the multiline option for the regex? I guess `/^role:([a-zA-Z]+)/i` would be a better fit (we don't care about a case-sensitive match, and I'd prefer `/` instead of `#` for the regex pattern for the sake of consistency with the rest of the library).